docs/guides/batch_norm.rst

```diff
@@ -0,0 +1,294 @@
+Using BatchNorm
+===============
+
+In this guide we will go through the details of using ``BatchNorm`` in models,
+in the process we will highlight some of the differences between code that uses
+``BatchNorm`` and code that does not.
+
+.. testsetup::
+
+  import flax.linen as nn
+  import jax.numpy as jnp
+  import jax
+  import optax
+  from typing import Any
+  from flax.core import FrozenDict
+
+Defining the model
+******************
+
+``BatchNorm`` is a Module that has different runtime behavior between training and
+inference. In other frameworks this behavior specified via mutable state, however
+in Flax we specify it explicitly via the ``use_running_average`` argument.
+A common pattern is to accept a ``train`` argument in the parent Module and use it to define
+``BatchNorm``'s ``use_running_average`` argument.
+
+.. codediff::
+  :title_left: regular code
+  :title_right: with BatchNorm
+  :sync:
+
+  class MLP(nn.Module):
+    @nn.compact
+    def __call__(self, x):
+      x = nn.Dense(features=4)(x)
+
+      x = nn.relu(x)
+      x = nn.Dense(features=1)(x)
+      return x
+
+  ---
+  class MLP(nn.Module):
+    @nn.compact
+    def __call__(self, x, train: bool): #!
+      x = nn.Dense(features=4)(x)
+      x = nn.BatchNorm(use_running_average=not train)(x) #!
+      x = nn.relu(x)
+      x = nn.Dense(features=1)(x)
+      return x
+
+Once the model is created initializing is very similar, the additional
+``train`` argument is passed to ``init`` to get back the ``variables`` structure.
+
+The ``batch_stats`` collection
+******************************
+
+Apart from the ``params`` collection, ``BatchNorm``
```

Comment:
It may be good to link to our [Variable collections documentation](https://flax.readthedocs.io/en/latest/api_reference/flax.linen.html#module-flax.core.variables) for people unfamiliar with this term.